api/app.py

```diff
@@ -42,25 +48,63 @@
 PalaceXrayProfiler.configure(app)
 
 
-@app.before_first_request
-def initialize_database(autoinitialize=True):
-    testing = "TESTING" in os.environ
+def initialize_application() -> Flask:
+
+    initialize_database()
+    initialize_circulation_manager()
+    initialize_admin()
+    return app
 
+
+def initialize_database(autoinitialize=True, is_testing=False):
```

Comment:
I'm torn about adding a `is_testing` parameter here. On one hand, It is *really* nice to get rid of:
```python
    testing = "TESTING" in os.environ
```
And I think we should be cleaning things like this up as we go.

My concern though is that we still use that environment variables in other places in the code base. Even [in other functions called from this one](https://github.com/ThePalaceProject/circulation/blob/main/core/config.py#L311). If you do a search for "TESTING", you'll find a number of places we use that check to decide if we are in test mode or not.

Adding this parameter here, means that anyone calling this function in test needs to remember both to set the "TESTING" env var (otherwise they will get a connection to the [wrong DB URL](https://github.com/ThePalaceProject/circulation/blob/main/core/config.py#L311)) and also set the `is_testing` parameter to get consistent behavior. Which feels like a bit of a trap we are laying for our future selves.

Maybe getting rid of the "TESTING" env var here needs to happen in sync with removing it in other places in the code in a future PR?

Reply:
That's valid.  I'll revert that change and create a ticket to remove "TESTING" from the codebase.